akka-docs/rst/java/code/docs/stream/javadsl/cookbook/RecipeReduceByKeyTest.java

```diff
@@ -51,12 +51,14 @@ public void work() throws Exception {
 
         //#word-count
         final int MAXIMUM_DISTINCT_WORDS = 1000;
-        
+
         final Source<Pair<String, Integer>, BoxedUnit> counts = words
-            // split the words into separate streams first
+          // split the words into separate streams first
           .groupBy(MAXIMUM_DISTINCT_WORDS, i -> i)
+          //transform each element to pair with number of words in it
+          .map(i -> new Pair<String, Integer>(i, 1))
           // add counting logic to the streams
-          .fold(new Pair<>("", 0), (pair, elem) -> new Pair<>(elem, pair.second() + 1))
+          .<Pair<String, Integer>> reduce((left, right) -> new Pair<>(left.first(), left.second() + right.second()))
```

Comment:
I think that this is because Java can not really understand [T >: Out]:  ```def reduce[T >: Out](f: function.Function2[T, T, T]): SubSource[T, Mat] ``` 
follow API works without generic in Java: ```def reduce[T >: Out](f: function.Function2[Out, Out, T]): SubSource[T, Mat] ``` 
However List.reduce is ```def reduce[A1 >: A](op: (A1, A1) ⇒ A1): A1 ```